src/lib/db.c

```diff
@@ -2324,7 +2324,26 @@ static CK_RV db_create(char *path, size_t len) {
 
 static FILE *take_lock(const char *path, char *lockpath) {
 
-    unsigned l = snprintf(lockpath, PATH_MAX, "%s%s", path, ".lock");
+    unsigned l;
+
+    char *env_lock = getenv("PKCS11_SQL_LOCK");
+
+    if (env_lock) {
+        /*
+         * lock file shall be "PKCS11_SQL_LOCK" + path + ".lock", but
+         * path's '/' will be substituted by '_'.
+         */
+        char path_alt[PATH_MAX];
```

Comment:
Let me be clear here, im either looking for a calloc or to use lockpath buffer in place....

Reply:
thanks @williamcroberts , I do understand your proposals, but please see my comment: I feel we could keep it simple.

Reply:
done with the commit 937af5884c5ea2eebdc3bb46ee58c19930abcf79